include/interfaceinteg.hpp

```diff
@@ -11,191 +11,237 @@
-   // offsets are used only for jmat, to determine the lower-triangular part.
-   static void AssembleBlock(
-      const int dim, const int row_ndofs, const int col_ndofs,
-      const int row_offset, const int col_offset, const double jmatcoef,
-      const Vector &row_shape, const Vector &col_shape, const Vector &col_dshape_dnM,
-      DenseMatrix &elmat, DenseMatrix &jmat);
-};
-
-class InterfaceDGNormalFluxIntegrator : public InterfaceNonlinearFormIntegrator
-{
-private:
-   int dim;
-   int order;
-   int p;
-
-   int trial_dof1, trial_dof2, test_dof1, test_dof2;
-   int trial_vdof1, trial_vdof2;
-
-   double w, wn;
-   int i, j, idof, jdof, jm;
-
-   Vector nor, wnor;
-   Vector shape1, shape2;
-   // Vector divshape;
-   Vector trshape1, trshape2;
-   // DenseMatrix vshape1, vshape2;
-   // Vector vshape1_n, vshape2_n;
-
-public:
-   InterfaceDGNormalFluxIntegrator() {};
-   virtual ~InterfaceDGNormalFluxIntegrator() {};
-
-   virtual void AssembleInterfaceMatrix(const FiniteElement &trial_fe1,
-                                       const FiniteElement &trial_fe2,
-                                       const FiniteElement &test_fe1,
-                                       const FiniteElement &test_fe2,
-                                       FaceElementTransformations &Trans1,
-                                       FaceElementTransformations &Trans2,
-                                       Array2D<DenseMatrix*> &elmats);
-
-   using InterfaceNonlinearFormIntegrator::AssembleInterfaceMatrix;
-};
-
-class InterfaceDGTemamFluxIntegrator : public InterfaceNonlinearFormIntegrator
-{
-private:
-   int dim, ndofs1, ndofs2, nvdofs1, nvdofs2;
-   double w, un;
-   Coefficient *Q{};
-
-   Vector nor, shape1, shape2, u1, u2, flux;
-   DenseMatrix udof1, udof2, elv1, elv2;
-   DenseMatrix elmat_comp11, elmat_comp12, elmat_comp21;
-
-public:
-   InterfaceDGTemamFluxIntegrator(Coefficient &q) : Q(&q) {};
-   virtual ~InterfaceDGTemamFluxIntegrator() {};
-
-   virtual void AssembleInterfaceVector(const FiniteElement &el1,
-                                       const FiniteElement &el2,
-                                       FaceElementTransformations &Tr1,
-                                       FaceElementTransformations &Tr2,
-                                       const Vector &elfun1, const Vector &elfun2,
-                                       Vector &elvect1, Vector &elvect2);
-
-   virtual void AssembleInterfaceGrad(const FiniteElement &el1,
-                                      const FiniteElement &el2,
-                                      FaceElementTransformations &Tr1,
-                                      FaceElementTransformations &Tr2,
-                                      const Vector &elfun1, const Vector &elfun2,
-                                      Array2D<DenseMatrix*> &elmats);
-};
+   class InterfaceNonlinearFormIntegrator : virtual public HyperReductionIntegrator
```

Comment:
Although we don't enforce any stylization yet, we'd like to keep the current indentation style. This is consistent with MFEM. Please recover the indentation throughout the code.